core/steps/cloudinit/disk_mount.go

```diff
@@ -0,0 +1,142 @@
+package cloudinit
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+
+	"github.com/sirupsen/logrus"
+	"gopkg.in/yaml.v2"
+
+	"github.com/weaveworks-liquidmetal/flintlock/client/cloudinit"
+	"github.com/weaveworks-liquidmetal/flintlock/client/cloudinit/userdata"
+	"github.com/weaveworks-liquidmetal/flintlock/core/models"
+	"github.com/weaveworks-liquidmetal/flintlock/pkg/log"
+	"github.com/weaveworks-liquidmetal/flintlock/pkg/planner"
+)
+
+func NewDiskMountStep(vm *models.MicroVM) planner.Procedure {
+	return &diskMountStep{
+		vm: vm,
+	}
+}
+
+type diskMountStep struct {
+	vm *models.MicroVM
+}
+
+// Name is the name of the procedure/operation.
+func (s *diskMountStep) Name() string {
+	return "cloudinit_disk_mount"
+}
+
+func (s *diskMountStep) ShouldDo(ctx context.Context) (bool, error) {
+	logger := log.GetLogger(ctx).WithFields(logrus.Fields{
+		"step": s.Name(),
+	})
+	logger.Debug("checking if procedure should be run")
+
+	if !s.vm.Spec.AdditionalVolumes.HasMountableVolumes() {
+		return false, nil
+	}
+
+	for _, vol := range s.vm.Spec.AdditionalVolumes {
+		if vol.MountPoint == "" {
+			continue
+		}
+
+		status := s.vm.Status.Volumes[vol.ID]
+
+		if status == nil || status.Mount.Source == "" {
+			return true, nil
+		}
+	}
+
+	vendorData, err := s.getVendorData()
+	if err != nil {
+		return false, fmt.Errorf("getting vendor data: %w", err)
+	}
+	if vendorData == nil {
+		return true, nil
+	}
+
+	for _, vol := range s.vm.Spec.AdditionalVolumes {
+		if vol.MountPoint == "" {
+			continue
+		}
+
+		if !vendorData.HasMountByMountPoint(vol.MountPoint) {
```

Comment:
Flintlock still does not support vm updates, right?

---
(if we support that)

Testing only the `MountPoint` can lead to a state where we don't update a vm as it looks up-to-date, but it's not.

Lets' say, originally we had this extra mount list:
```
volume1.vol -> /home/
volume2.vol -> /opt/device-stats
```

1st case) We realize we messed up the volumes and we try to update
```
volume1.vol -> /opt/device-stats
volume2.vol -> /home/
```

2nd case) We completely change the source volumes.
```
volume3.vol -> /home/
volume4.vol -> /opt/device-stats
```

If nothing else changes in the vm spec, no other checks will force a recreate.

Reply:
I feel we don't support any (or at least just very limited) updating of vms, no? The `ShouldDo` steps check for surface level things on most attributes. We don't really have any idempotency or 'smart' state mechanism.